lib/deploy.js

```diff
@@ -32,7 +32,20 @@ module.exports = promisify(function(config, callback) {
   var ghPagesConfig = {};
   var rootDir = config.rootDir ? config.rootDir : '.';
   var cloneDir = config.cloneDir ? config.cloneDir : null;
-  var commitMessage = config.message ? config.message : childProcess.execSync('git log --format=%B -n 1').toString().trim();
+
+  var commitMessage = config.message || (function() {
+    var spawn = childProcess.spawnSync('git', ['log', '--format=%B', '-n', '1']);
+    var errorText = spawn.stderr.toString().trim();
+
+    if (errorText) {
+      gutil.log('Fatal error from `git log`.  You must have one commit before deploying.');
+      throw new Error(errorText);
+      callback(spawn.stderr);
```

Comment:
Here's the difference in the gulp output, if you use *throw*, between the two versions I posted above:
```
[18:01:09] Using gulpfile ~/Documents/workspace/oghliner/asd/gulpfile.js
[18:01:09] Starting 'deploy'...
[18:01:09] Fatal error from `git log`.  You must have one commit before deploying.
```

```
[18:01:48] Using gulpfile ~/Documents/workspace/oghliner/asd/gulpfile.js
[18:01:48] Starting 'deploy'...
[18:01:48] Fatal error from `git log`.  You must have one commit before deploying.
[18:01:48] 'deploy' errored after 8.57 ms
[18:01:48] Error: fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).

    at formatError (/usr/local/lib/node_modules/gulp/bin/gulp.js:169:10)
    at Gulp.<anonymous> (/usr/local/lib/node_modules/gulp/bin/gulp.js:195:15)
    at Gulp.emit (events.js:107:17)
    at Gulp.Orchestrator._emitTaskDone (/home/marco/Documents/workspace/oghliner/asd/node_modules/gulp/node_modules/orchestrator/index.js:264:8)
    at /home/marco/Documents/workspace/oghliner/asd/node_modules/gulp/node_modules/orchestrator/index.js:275:23
    at finish (/home/marco/Documents/workspace/oghliner/asd/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:21:8)
    at /home/marco/Documents/workspace/oghliner/asd/node_modules/gulp/node_modules/orchestrator/lib/runTask.js:45:4
    at tryCallOne (/home/marco/Documents/workspace/oghliner/oghliner/node_modules/promisify-node/node_modules/nodegit-promise/lib/core.js:37:12)
    at /home/marco/Documents/workspace/oghliner/oghliner/node_modules/promisify-node/node_modules/nodegit-promise/lib/core.js:103:15
    at flush (/home/marco/Documents/workspace/oghliner/oghliner/node_modules/promisify-node/node_modules/nodegit-promise/node_modules/asap/raw.js:50:29)
```